grammar.js

```diff
@@ -249,6 +268,25 @@ module.exports = grammar(C, {
 
     // Declarations
 
+    _typedef_type_specifier: $ => choice(
+      $.macro_type_specifier,
+      alias($._typedef_sized_type_specifier, $.sized_type_specifier),
+      $.struct_specifier,
+      $.union_specifier,
+      $.enum_specifier,
+      $.class_specifier,
+      $.sized_type_specifier,
+      $.primitive_type,
+      $.template_type,
+      $.dependent_type,
+      $.placeholder_type_specifier,
+      $.decltype,
+      prec.right(choice(
+        alias($.qualified_type_identifier, $.qualified_identifier),
+        $._type_identifier,
+      )),
+    ),
```

Comment:
What is this for?

Reply:
I reworked how typedef worked to fix cases like `typedef unsigned uint;`, but I reworked it again to get rid of that and make it simpler, https://github.com/tree-sitter/tree-sitter-c/pull/152 for reference. We can update CPP to use git head for tree-sitter-c and port the changes